rust/src/nfs/nfs.rs

```diff
@@ -85,6 +86,22 @@ static mut ALPROTO_NFS: AppProto = ALPROTO_UNKNOWN;
  * Transaction lookup.
  */
 
+#[derive(AppLayerFrameType)]
+pub enum NFSFrameType {
+    RPCPdu,
+    RPCHdr,
+    RPCData,
+    RPCCreds, // for rpc calls
+
+    NFSPdu,
```

Comment:
should we also split out nfs2 and nfs3 here? I think splitting out nfs4 makes total sense, the format is very different. How do you see this for 2 vs 3?

Reply:
As I went about this I had them separated at first, but then I found that they are mostly the same, some records --a good deal of them-- are even identical. As for the format, they don't have a distinguished format, there is no wrapper field that could be extracted out as a frame as in the `nfs4.ops` frame that holds all `ops` data, so for `NFS2 | NFS3` its just a blob of data --pdu-- with fields according to the records, and a `Status` for the responses.

The way I see it: If its important from the perspective of having versions show up in the frames wrt rule writing and the like, then yes we could separate it, other than that a generic `nfs` would do the work and reduce repetitiveness.

Reply:
think this makes sense. It's a lot like how in the ssl/tls parser we treat ssl3/tls1+ all the same, while doing something separate for ssl2. Thanks.